runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -4,12 +4,30 @@ local Range = require('vim.treesitter._range')
 
 local ns = api.nvim_create_namespace('treesitter/highlighter')
 
----@alias vim.treesitter.highlighter.Iter fun(end_line: integer|nil): integer, TSNode, vim.treesitter.query.TSMetadata, TSQueryMatch
+--- @class vim.treesitter.highlighter.Iter.ret
```

Comment:
Not a blocker, but given this is a structure for the public API `|vim.treesitter.get_highlight()|` it would deserve a better class name rather than internal class name, say `vim.treesitter.highlighter.Info` or `vim.treesitter.HighlightInfo`, etc. Though I don't think these two are good enough.